### DISCUSSION TITLE
Qt Websocket

### DISCUSSION THREAD

**Discussion by dridk** [ecorm/cppwamp#124] [General]
Could you make it work with Qt which support now Qwebsocket ? Maybe a new repository

**Comment by ecorm** [comment]
The question of providing a "binding" to Qt has already been raised in #62. When I eventually get around to adding a WebSocket-based transport, I plan us using websocketpp (https://github.com/zaphoyd/websocketpp), which is also based on Boost.Asio.

It's too messy/difficult to integrate a library based on boost::asio::io_service with Qt's event loop. Of course, anyone is welcome to fork this project and attempt such a marriage (if its even possible), as long as this project's code is redistributed under the same license.

In my option, a Qt-based WAMP client library would be best written from scratch using all of Qt's facilities and its idioms. This project (as well as Autobahn|CPP), could serve as a "model" to such a new library.

**Comment by estan** [comment]
I second @ecorm's opinion. In fact, I have a few more credits at university this Fall, before I have my degree, and I've decided to spend them by taking two "individual courses" where the student is free to design his own project. My project suggestion is precisely to build a "Qt native" WAMP client library and I'm waiting for teachers/examiner to say whether the project is OK (might be too little "theoretical computer science" for them).

But should I start work on such a library, I'll comment here and on my original #62.

As a side note: One current limitation of QWebSocket is that it has no support for WebSocket subprotocol negotiation, but this has been reported as a bug and will be fixed in some future Qt version. Until then, once can always turn off the negotiation in Crossbar (in effect forcing one subprotocol or the other).

**Comment by dridk** [comment]
Here is the request to the Qt developer :

https://bugreports.qt.io/browse/QTBUG-38742

You can comment and join the project

2015-08-27 15:42 GMT+02:00 Elvis Stansvik notifications@github.com:

> I second @ecorm https://github.com/ecorm's opinion. In fact, I have a
> few more credits at university this Fall, before I have my degree, and I've
> decided to spend them by taking two "individual courses" where the student
> is free to design his own project. My project suggestion is precisely to
> build a "Qt native" WAMP client library and I'm waiting for
> teachers/examiner to say whether the project is OK (might be too little
> "theoretical computer science" for them).
> 
> But should I start work on such a library, I'll comment here and on my
> original #62 https://github.com/ecorm/cppwamp/issues/62.
> 
> As a side note: One current limitation of QWebSocket is that it has no
> support for WebSocket subprotocol negotiation, but this has been reported
> as a bug and will be fixed in some future Qt version. Until then, once can
> always turn off the negotiation in Crossbar (in effect forcing one
> subprotocol or the other).
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ecorm/cppwamp/issues/74#issuecomment-135434515.

## 

## 

Sacha schutz
+33651571606
dridk.me
Fork me on github https://github.com/

**Comment by estan** [comment]
@dridk Yep. I did comment on that bug back in March, and also in June, but it seems it's quite low priority for the Qt devs. Due to working on my thesis project, I haven't had time to look into it myself. But I might do that if I get time and no other Qt dev gets to it first.

**Comment by ecorm** [comment]
Closing, as I am planning to use websocketpp, and not QWebSocket. We don't use Qt in our project, so I don't plan of providing Qt application integration support. Whatever spare time I have to enhance CppWAMP would go towards implementing more advanced spec features.

If someone wants to write some kind of Qt <--> CppWAMP adapter and needs minor changes to the CppWAMP library, then you may submit such a request as a GitHub issue.

If you wish to discuss an upcoming Qt-based WAMP client, you are welcome to use the [CppWAMP Google Group](https://groups.google.com/forum/#!forum/cppwamp).
